x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/pytorch/process/PyTorchResultProcessor.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ml.inference.pytorch.process;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.message.ParameterizedMessage;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.xpack.core.ml.inference.deployment.PyTorchResult;
+
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+public class PyTorchResultProcessor {
+
+    private static final Logger logger = LogManager.getLogger(PyTorchResultProcessor.class);
+
+    private final ConcurrentMap<String, PendingResult> pendingResults = new ConcurrentHashMap<>();
+
+    private final String deploymentId;
+    private volatile boolean isStopping;
+
+    public PyTorchResultProcessor(String deploymentId) {
+        this.deploymentId = Objects.requireNonNull(deploymentId);
+    }
+
+    public void process(PyTorchProcess process) {
+        try {
+            Iterator<PyTorchResult> iterator = process.readResults();
+            while (iterator.hasNext()) {
+                PyTorchResult result = iterator.next();
+                logger.debug(() -> new ParameterizedMessage("[{}] Parsed result with id [{}]", deploymentId, result.getRequestId()));
+                PendingResult pendingResult = pendingResults.get(result.getRequestId());
+                if (pendingResult == null) {
+                    logger.debug(() -> new ParameterizedMessage("[{}] no pending result for [{}]", deploymentId, result.getRequestId()));
```

Comment:
```suggestion
                    logger.warn(() -> new ParameterizedMessage("[{}] no pending result for [{}]", deploymentId, result.getRequestId()));
```

This is interesting enough to be `warn`

Reply:
This should only occur if the infer request timed out. That would indicate a throughput problem. I'm not sure we'd want to fill the log with those in that case as we could be getting an entry per document we're trying to apply inference to.